File: GeometryCalcKotlin/src/amdegregorio/geo/GeometryCalc.kt
package amdegregorio.geo

import kotlin.system.exitProcess

fun main(args : Array<String>) {
	var length:Int = 42
	var width:Int = 20
	
	if (args.size > 0) {
		try {
			if (args.size == 1) {
				length = args.get(0).toInt()
				width = args.get(0).toInt()
			} else {
				length = args.get(0).toInt()
				width = args.get(1).toInt()
			}
		} catch (e: NumberFormatException) {
			println("Usage: java -jar GeometryCalc.jar [length] [width]")
			println("  length: a whole number representing the length of a rectangle")
         println("  width: a whole number representing the width of a rectangle")
			exitProcess(1)
		}
	}
	
	var rectangle = Rectangle(length, width)
	println("Created rectangle: " + rectangle.toString())
   println("Area: " + rectangle.calculateArea())
   println("Perimeter: " + rectangle.calculatePerimeter())
	exitProcess(0)
}

File: amdegregorio.geometrycalc/src/amdegregorio/geometrycalc.clj
(ns amdegregorio.geometrycalc
  (:gen-class)
  (:require amdegregorio.rectangle)
  (:import [amdegregorio.rectangle Rectangle]))

;from https://stackoverflow.com/questions/5621279/in-clojure-how-can-i-convert-a-string-to-a-number
(defn parse-int [v] 
   (try 
     (Integer/parseInt (re-find #"^\d+" (.toString v))) 
     (catch NumberFormatException e 0)))

(defn -main
    [& args]
  ; Read the args and set up length and width
  ; if no command line args supplied, use default values
  ; if one command line arg sent - its a square use length for length and width
  (if (nth args 0) 
    (def length (parse-int (nth args 0)))
    (def length 42))
  (if (= (count args) 2)
    (def width (parse-int (nth args 1)))
    (if (= (count args) 1)
      (def width length)
      (def width 20)))
  ;create the rectangle and print it's vitals
  (def rectangle (Rectangle. length width))
  (println rectangle)
  (println(.calculateArea rectangle))
  (println(.calculatePerimeter rectangle))
)